### ISSUE TITLE
Intermittent test failure : org.apache.pulsar.broker.service.ReplicatorTest.testCloseReplicatorStartProducer

### ISSUE THREAD

**Issue by rdhabalia** [apache/pulsar#890]
[build](https://builds.apache.org/job/pulsar-pull-request/org.apache.pulsar$pulsar-broker/695/testReport/junit/org.apache.pulsar.broker.service/ReplicatorTest/testCloseReplicatorStartProducer/)

```
Error Message

Producer was not registered on the connection
Stacktrace

org.apache.pulsar.client.api.PulsarClientException: Producer was not registered on the connection
	at org.apache.pulsar.client.impl.ClientCnx.getPulsarClientException(ClientCnx.java:531)
	at org.apache.pulsar.client.impl.ClientCnx.handleError(ClientCnx.java:367)
	at org.apache.pulsar.common.api.PulsarDecoder.channelRead(PulsarDecoder.java:148)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:356)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:342)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:335)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:295)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:269)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:356)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:342)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:335)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:356)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:342)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:934)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:397)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:302)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:131)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)

```

**Comment by ivankelly** [comment]
Instead of fixing as it is now, this should be converted to an integration test first.

**Comment by merlimat** [comment]
Test was fixed in e4fc42da737f5d62bb4d19e2d306a1d7101ba9bf and 1921bd98f20991248892717bbcfc102545e7c49c
